buddy_alloc.h

```diff
@@ -383,7 +383,11 @@ struct buddy {
         ptrdiff_t main_offset;
     } arena;
     size_t buddy_flags;
+#ifndef __cplusplus
     unsigned char buddy_tree[];
+#else
+    unsigned char *buddy_tree;
```

Comment:
CPP's lack of FMAs, right ? Let me think about how it could be handled in a way that it won't require cpp ifdefs everywhere.